src/lib/airlock/airlock.js

```diff
@@ -0,0 +1,56 @@
+import { base } from '../airtable/airtable';
+import {
+  refreshUserData,
+  clearUserData,
+} from '../redux/userData';
+
+const AUTHENTICATION_ERR_STRING = 'AUTHENTICATION_REQUIRED';
+
+const signupUser = async (email, password, fullname, role) => {
+  try {
+    /* eslint-disable */
```

Comment:
Your sign up function currently has no way of evaluating if the person creating an account is giving a duplicate password. Also, your base.register will return nothing bc there is no await.

Here's what I would do:
`const signupUser = async (email, password, fullname, role) => {
  try {
    /* eslint-disable */
    const res = await base.register({
      username: email,
      password: password,
      fields: {
        user_type: role,
        display_name: fullname,
        approval: 'unapproved',
      },
    });
    console.log(res);
    //dummy return value for now do not return 1!!
    return 1;
  } catch (err) {
    return null;
  }
};`

And I would use res.body.success to evaluate if the signup was successful and return something based on that!